### PR TITLE
INTCMN-558 : New static methods for getting BlackDuckServerConfigBuilder

### PR DESCRIPTION
**Pull request by DMaxfield-BDS**
* Implement static methods: newApiBuilder() and newUserPassBuilder()
* Deprecate original static method: newBuilder()
* Add tests for 100% unit coverage of BlackDuckServerConfigBuilder

### CODE REVIEW

**Review by crowleySynopsys — APPROVED** [review]

File: src/test/java/com/synopsys/integration/blackduck/configuration/BlackDuckServerConfigBuilderTest.java [hunk 1]
```diff
@@ -140,12 +181,366 @@ public void testPopulatingExecutorService() throws Exception {
             executorServiceField.setAccessible(true);
             assertTrue(executorServiceField.get(blackDuckServerConfig) instanceof NoThreadExecutorService);
 
+            assertNotNull(blackDuckServerConfigBuilder.getExecutorService());
             blackDuckServerConfigBuilder.setExecutorService(executorService);
+            assertEquals(executorService, blackDuckServerConfigBuilder.getExecutorService().orElse(null));
             blackDuckServerConfig = blackDuckServerConfigBuilder.build();
             assertNotNull(executorServiceField.get(blackDuckServerConfig));
         } finally {
+            assert executorService != null;
```
Comment (JakeMathews) [inline src/test/java/com/synopsys/integration/blackduck/configuration/BlackDuckServerConfigBuilderTest.java:190]: Why is this a different type of assertion?

File: src/test/java/com/synopsys/integration/blackduck/configuration/BlackDuckServerConfigBuilderTest.java [hunk 2]
```diff
@@ -140,12 +181,366 @@ public void testPopulatingExecutorService() throws Exception {
...
             executorService.shutdownNow();
         }
     }
 
+    @Test
+    public void testNewBuilder() {
```
Comment (taikuukaits) [inline src/test/java/com/synopsys/integration/blackduck/configuration/BlackDuckServerConfigBuilderTest.java:196]: The method names of the tests are not super helpful. I think something like testNewBuilderIncludesAllKeys() would be better. I also didn't see a test for 'provides username password to newApiTokenBuilder() and it is not valid', ie the detect use case.

**Review by taikuukaits — APPROVED** [review]

**Review by JakeMathews — APPROVED** [review]

File: src/test/java/com/synopsys/integration/blackduck/configuration/BlackDuckServerConfigBuilderTest.java (hunk 1 above)
Comment (DMaxfield-BDS) [reply src/test/java/com/synopsys/integration/blackduck/configuration/BlackDuckServerConfigBuilderTest.java:182]: shutdownNow(), called on the next line, can throw NPE. Would you say to change this to an if check, or just remove the check completely?

Comment (JakeMathews) [reply src/test/java/com/synopsys/integration/blackduck/configuration/BlackDuckServerConfigBuilderTest.java:182]: My confusion was in it not being ‘Assertions.assertNull’, but if this is for the test cleanup it’s fine

**Review by crowleySynopsys — APPROVED** [review]

**Review by JakeMathews — APPROVED** [review]

**Pull request merged**